addons/website_crm/models/crm_lead.py

```diff
@@ -51,7 +51,8 @@ def website_form_input_filter(self, request, values):
                             request.website.crm_default_team_id.id
         values['user_id'] = values.get('user_id') or \
                             request.website.crm_default_user_id.id
-        values['type'] = 'lead' if self.with_user(SUPERUSER_ID).env['res.users'].has_group('crm.group_use_lead') else 'opportunity'
+        team = self.with_user(SUPERUSER_ID).env['crm.team'].search([('id', '=', values['team_id'])])
```

Comment:
Is `with_user(SUPERUSER_ID)` required?

Otherwise, LGTM

Reply:
Hello, yes. I tried without at first but the test tour does not pass. I think it is because it cannot read into "crm.team" if portal users (most leads/opportunities)

Reply:
Beware, not sure you always have a team_id value .. and can't you browse as sudo ?

Something like
```
if values.get('team_id'):
    type = 'lead' if self.env['crm.team'].sudo().browse(values['team_id']).use_leads else 'opportunity'
else:
    type = self.with_user(SUPERUSER_ID).env['res.users'].has_group('crm.group_use_lead') else 'opportunity'
```